cgen: error: , expected

### Describe the bug

from discord https://discordapp.com/channels/592103645835821068/592294828432424960/1060916264266117270

![image](https://user-images.githubusercontent.com/63821277/211026815-74caca52-fec7-4ce6-80ee-09b9847f8db6.png)


### Expected Behavior

?

### Current Behavior

```bash
==================
/tmp/v_1000/16861.5334599694026439295.tmp.c:12555: error: , expected
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

### Reproduction Steps

```v
fn main() {
    myu16 := u16(0)
    size_of_u16 := sizeof(u16)

    mut asu8 := &u8(unsafe{malloc(size_of_u16)})

    asu8 = &[myu16]! // look this

    assert asu8.vbytes(int(size_of_u16)) == [u8(0),0]
}
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.3.2 3b594d6

### Environment details (OS name and version, etc.)

Ubuntu

--- a/vlib/v/gen/c/array.v
+++ b/vlib/v/gen/c/array.v
@@ -25,6 +25,9 @@ fn (mut g Gen) array_init(node ast.ArrayInit, var_name string) {
 	len := node.exprs.len
 	if array_type.unaliased_sym.kind == .array_fixed {
 		g.fixed_array_init(node, array_type, var_name)
+		if is_amp {
+			g.write(')')
+		}
 	} else if len == 0 {
 		// `[]int{len: 6, cap:10, init:22}`
 		g.array_init_with_fields(node, elem_type, is_amp, shared_styp, var_name)
@@ -60,11 +63,11 @@ fn (mut g Gen) array_init(node ast.ArrayInit, var_name string) {
 			}
 		}
 		g.write('}))')
-	}
-	if g.is_shared {
-		g.write('}, sizeof(${shared_styp}))')
-	} else if is_amp {
-		g.write(')')
+		if g.is_shared {
+			g.write('}, sizeof(${shared_styp}))')
+		} else if is_amp {
+			g.write(')')
+		}
 	}
 }
 

--- a/vlib/v/gen/c/array.v
+++ b/vlib/v/gen/c/array.v
@@ -60,11 +60,11 @@ fn (mut g Gen) array_init(node ast.ArrayInit, var_name string) {
 			}
 		}
 		g.write('}))')
-		if g.is_shared {
-			g.write('}, sizeof(${shared_styp}))')
-		} else if is_amp {
-			g.write(')')
-		}
+	}
+	if g.is_shared {
+		g.write('}, sizeof(${shared_styp}))')
+	} else if is_amp {
+		g.write(')')
 	}
 }
 

--- a/vlib/v/tests/assert_with_array_ref_test.v
+++ b/vlib/v/tests/assert_with_array_ref_test.v
@@ -1,0 +1,10 @@
+fn test_main() {
+	myu16 := u16(50)
+	size_of_u16 := sizeof(u16)
+
+	mut asu8 := unsafe { &u8(malloc(size_of_u16)) }
+
+	asu8 = &[myu16]! // look this
+
+	assert unsafe { asu8.vbytes(int(size_of_u16)) } == [u8(50), 0]
+}

